src/components/nabla/Swap/To.tsx

```diff
@@ -74,14 +74,25 @@ export function To({
           type="button"
         >
           <span className="mr-1 h-full rounded-full bg-[rgba(0,0,0,0.15)] p-px">
-            <img src={getIcon(toToken?.symbol)} alt={toToken?.name} className="h-full w-auto" />
+            <img src={getIcon(toToken?.symbol)} alt={toToken?.name} className="w-auto h-full" />
           </span>
           <strong className="font-bold">{toToken?.symbol || 'Select'}</strong>
-          <ChevronDownIcon className="ml-px inline h-4 w-4" />
+          <ChevronDownIcon className="inline w-4 h-4 ml-px" />
         </Button>
       </div>
-      <div className="mt-1 flex items-center justify-between text-neutral-500 dark:text-neutral-300">
-        <div className="mt-px text-sm">{toToken ? <NablaTokenPrice address={toToken.id} fallback="$ -" /> : '$ -'}</div>
+      <div className="flex items-center justify-between mt-1 text-neutral-500 dark:text-neutral-300">
+        <div className="mt-px text-sm">
+          {toToken ? (
+            <NablaTokenPrice
+              formatByAmount={true}
+              currentTokenAmount={Number(toAmountQuote.data?.amountOut.approximateStrings.atLeast4Decimals) || 0}
```

Comment:
Let's generally avoid `number` for financial applications and only use the `Big.js` library or native `bigint`.

```suggestion
              currentTokenAmount={Big(toAmountQuote.data?.amountOut.approximateStrings.atLeast4Decimals ?? "0")}
```